api-inference-community/tests/test_nlp.py

```diff
@@ -49,16 +49,149 @@ def test_sentence_similarity(self):
+        )
+        self.assertEqual(processed_params, {})
+        self.assertEqual(inputs, normalized_inputs)
+
+    def test_invalid_question(self):
+        query = "How many stars does the transformers repository have?"
+        table = "Invalid table"
+        inputs = {"query": query, "table": table}
+        bpayload = json.dumps({"inputs": inputs}).encode("utf-8")
+        with self.assertRaises(ValidationError):
+            normalize_payload_nlp(bpayload, "table-question-answering")
+
+    def test_invalid_query(self):
+        query = {"not a": "query"}
+        table = {
+            "Repository": ["Transformers", "Datasets", "Tokenizers"],
+            "Stars": ["36542", "4512", "3934"],
+        }
+        inputs = {"query": query, "table": table}
+        bpayload = json.dumps({"inputs": inputs}).encode("utf-8")
+        with self.assertRaises(ValidationError):
+            normalize_payload_nlp(bpayload, "table-question-answering")
+
+    def test_no_table(self):
+        query = "How many stars does the transformers repository have?"
+        inputs = {
+            "query": query,
+        }
+        bpayload = json.dumps({"inputs": inputs}).encode("utf-8")
+        with self.assertRaises(ValidationError):
+            normalize_payload_nlp(bpayload, "table-question-answering")
+
+    def test_no_query(self):
+        table = {
+            "Repository": ["Transformers", "Datasets", "Tokenizers"],
+            "Stars": ["36542", "4512", "3934"],
+        }
+        inputs = {"table": table}
+        bpayload = json.dumps({"inputs": inputs}).encode("utf-8")
+        with self.assertRaises(ValidationError):
+            normalize_payload_nlp(bpayload, "table-question-answering")
+
+
+class SummarizationValidationTestCase(TestCase):
+    def test_no_params(self):
+        bpayload = json.dumps({"inputs": "whatever"}).encode("utf-8")
+        normalized_inputs, processed_params = normalize_payload_nlp(
+            bpayload, "summarization"
+        )
+        self.assertEqual(processed_params, {})
+        self.assertEqual(normalized_inputs, "whatever")
+
+    def test_valid_min_length(self):
+        params = {"min_length": 10}
+        bpayload = json.dumps({"inputs": "whatever", "parameters": params}).encode(
+            "utf-8"
+        )
+        normalized_inputs, processed_params = normalize_payload_nlp(
+            bpayload, "summarization"
+        )
+        self.assertEqual(processed_params, params)
+        self.assertEqual(normalized_inputs, "whatever")
+
+    def test_max_length(self):
+        params = {"max_length": 10}
+        bpayload = json.dumps({"inputs": "whatever", "parameters": params}).encode(
+            "utf-8"
+        )
+        normalized_inputs, processed_params = normalize_payload_nlp(
+            bpayload, "summarization"
+        )
+        self.assertEqual(processed_params, params)
+        self.assertEqual(normalized_inputs, "whatever")
+
+    @parameterized.expand(
```

Comment:
I find those quite hard to read and quite error-prone. I've seen quite a few mistakes in transformers tests by over parametrization.

What's this issue with simply writing those explicitly ?
Another problem with the current implementation is that it's impossible to test parameters coupling ('min_length '> 'max_length' for instance.

Reply:
Fixed. I also added a test for the last point you mentioned.